File: backend/app/core/phone.py
import re


def normalize_ru_phone(s: str) -> str | None:
    """Парсит ввод (8..., +7..., 7..., 9...) в вид +7XXXXXXXXXX или None."""
    if not s or not str(s).strip():
        return None
    d = re.sub(r"\D", "", s.strip())
    if len(d) == 11 and d[0] == "8":
        d = "7" + d[1:]
    if len(d) == 10:
        d = "7" + d
    if len(d) == 11 and d[0] == "7":
        return f"+{d}"
    return None

File: mobile-app/src/utils/phone.ts
/** Как на бэкенде: +7XXXXXXXXXX или null ([web/app/utils/phone.ts](web/app/utils/phone.ts)) */
export function normalizeRuPhone(s: string): string | null {
  if (!s?.trim()) return null;
  const d = s.replace(/\D/g, "");
  let n = d;
  if (n.length === 11 && n[0] === "8") n = "7" + n.slice(1);
  if (n.length === 10) n = "7" + n;
  if (n.length === 11 && n[0] === "7") return `+${n}`;
  return null;
}
